src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -256,11 +256,15 @@ class Config {
   using OopT = std::conditional_t<oops == oop_kind::NARROW, narrowOop, oop>;
 
   static freeze_result freeze(JavaThread* thread, intptr_t* const sp) {
-    return freeze_internal<SelfT, false>(thread, sp);
+    freeze_result res = freeze_internal<SelfT, false>(thread, sp);
+    JFR_ONLY(assert((res == freeze_ok) || res == thread->last_freeze_fail_result(), "freeze failure not set"));
```

Comment:
Sorry, I forgot to add parenthesis to the 2nd comparison...

```suggestion
    JFR_ONLY(assert((res == freeze_ok) || (res == thread->last_freeze_fail_result()), "freeze failure not set"));
```